### ISSUE TITLE
Users can LOSE WORK when overlapping onScore blocks

### ISSUE THREAD

**Issue by BenVillalobos** [microsoft/pxt#9402]
Repro notes in https://arcade.makecode.com/S84683-03204-06431-95554

## Context
Users can **lose work** when overwriting an `onScore` call to the same score as a previous one.

1. Create multiple on-score blocks in blockly
2. Invalidate one in JS by setting one `onScore` call to the same score as another
3. Back in blockly editor the original onScore block is invalidated.
4. Go back to JS and see that the code no longer exists.
5. Sad user lost valuable work :(

I tried inspecting the html items and I can't find any way to recover the dead code.
![image](https://user-images.githubusercontent.com/4691428/221953427-94fe55cd-845e-4711-911c-c0b0b4774f76.png)

## Important Question
Is this the behavior of our block-invalidation logic? Can the user lose work on _any_ block invalidation that follows this flow?

**Comment by jwunderl** [comment]
If you drag this into the on start does it live again? it looks like it should as it's just a typescript block?

**Comment by BenVillalobos** [comment]
It does, which took me a bit of time to figure out since the type of block changed. I assume this behavior is expected then? I couldn't get the simulator to run until copy/pasting the code so it could replace the newly-formed JS block with the blockly-block

**Comment by jwunderl** [comment]
it's semi expected, though I would say the intention of https://github.com/microsoft/pxt/pull/8953 is probably that it gets stuck in an on start instead of just a free floating block @riknoll ?

**Comment by riknoll** [comment]
yeah, and i could have sworn i added a test case to specifically test that it did. maybe not, though!

**Comment by BenVillalobos** [comment]
Renamed the issue to reflect the needed change.
